source/pdo_sqlsrv/pdo_dbh.cpp

```diff
@@ -96,18 +96,18 @@ const stmt_option PDO_STMT_OPTS[] = {
 // boolean connection string
 struct pdo_bool_conn_str_func 
 {
-    static void func( connection_option const* option, zval* value, sqlsrv_conn* /*conn*/, std::string& conn_str TSRMLS_DC );
+    static void func( _In_ connection_option const* option, _Inout_ zval* value, sqlsrv_conn* /*conn*/, _Out_ std::string& conn_str TSRMLS_DC );
```

Comment:
pdo_bool_conn_str_func::func calls core_str_zval_is_true(value) (line 1561), which strips any whitespaces at the end and adjust the string length of the zend_string pointed to by value (core_conn (line 827))